pubsub/consumer.go

```diff
@@ -233,3 +233,24 @@ func (c *Consumer[Request, Response]) SetResult(ctx context.Context, messageID s
 	}
 	return nil
 }
+
+func (c *Consumer[Request, Response]) SetError(ctx context.Context, messageID string, error string) error {
+	resp, err := json.Marshal(error)
+	if err != nil {
+		return fmt.Errorf("marshaling result: %w", err)
+	}
+	errorKey := ErrorKeyFor(c.StreamName(), messageID)
+	log.Debug("consumer: setting error", "cid", c.id, "msgIdInStream", messageID, "errorKeyInRedis", errorKey)
+	acquired, err := c.client.SetNX(ctx, errorKey, resp, c.cfg.ResponseEntryTimeout).Result()
+	if err != nil || !acquired {
```

Comment:
can we return a predefined error if `!acquired`? 
e.g. 

```
var AlreadySetError = errors.New("redis key already set")
...
func SetError() {
    ...
    if !acquired && err == nil {
        err = AlreadySetError
    }
    if err != nil {
	return fmt.Errorf("setting error for message with message-id in stream: %v, error: %w", messageID, err)
    }
    ...
}
```
and then `SetError` caller would be able to check returned error against this error and know if `SetError` failed because of some real error or because of some other consumer setting the key first.

That will be useful, e.g. when we'll want to call `SetError` for a message that was retried too many times, so we could only print an error log when there is real error and not when another customer set the error first. 

And also it should be nicer to log
 `"setting error for message with message-id in stream: %v, error: redis key already set"` instead of
 `"setting error for message with message-id in stream: %v, error: nil"`


We can also add a similar error in `SetResult` method.